File: Python/Operators/LogicalOperator.py
# Logical Operators

a = True
b = False

# Logical AND
logical_and = a and b
# True AND False results in False
print("Logical AND:", logical_and)

# Logical OR
logical_or = a or b
# True OR False results in True
print("Logical OR:", logical_or)

# Logical NOT
logical_not_a = not a
# NOT True results in False
print("Logical NOT (a):", logical_not_a)

logical_not_b = not b
# NOT False results in True
print("Logical NOT (b):", logical_not_b)

# Combining Logical Operators
combined = (a and b) or (not a and not b)
# (True AND False) OR (NOT True AND NOT False)
# False OR False results in False
print("Combined Logical Operators:", combined)

File: Lua/Operators/LogicalOperator.lua
-- Logical Operators

-- Variables for logical operations
local a = true
local b = false

-- 1. Logical AND (and)
local andResult = a and b
print("Logical AND (a and b):", andResult)  -- Output: Logical AND (a and b): false

-- 2. Logical OR (or)
local orResult = a or b
print("Logical OR (a or b):", orResult)  -- Output: Logical OR (a or b): true

-- 3. Logical NOT (not)
local notResultA = not a
local notResultB = not b
print("Logical NOT (not a):", notResultA)  -- Output: Logical NOT (not a): false
print("Logical NOT (not b):", notResultB)  -- Output: Logical NOT (not b): true

-- Using logical operators in conditional statements
local x = 15
local y = 25

-- 4. Logical AND in if statement
if x < 20 and y > 20 then
    print("Both conditions are true (x < 20 and y > 20)")  -- Output: Both conditions are true (x < 20 and y > 20)
else
    print("One or both conditions are false")
end

-- 5. Logical OR in if statement
if x > 20 or y > 20 then
    print("At least one condition is true (x > 20 or y > 20)")  -- Output: At least one condition is true (x > 20 or y > 20)
else
    print("Both conditions are false")
end

-- 6. Logical NOT in if statement
if not (x > 20) then
    print("Condition is false (not (x > 20))")  -- Output: Condition is false (not (x > 20))
else
    print("Condition is true")
end

-- Combining logical operators
local z = 30

-- 7. Combining AND, OR, and NOT
if (x < y or y < z) and not (x > z) then
    print("Complex condition is true ((x < y or y < z) and not (x > z))")  -- Output: Complex condition is true ((x < y or y < z) and not (x > z))
else
    print("Complex condition is false")
end
